### DISCUSSION TITLE
Add force charge/discharge to H3

### DISCUSSION THREAD

**Discussion by Vato9394** [nathanmarlor/foxess_modbus#615] [Ideas]
I have an Foxess H3 and the ability to force charge/discharge was removed in a recent update see:
https://github.com/nathanmarlor/foxess_modbus/issues/569#issuecomment-2002580681

I want to troubleshoot together with you @canton7 if you have time for this to possibly enable this feature for the H3.

**Comment by canton7** [comment]
Do you have an H3 or an AIO-H3?

**Reply by Vato9394** [reply]
AIO-H3

**Reply by canton7** [reply]
Interesting. So another user (@PaRez280) reported that they were getting errors when I tried to read the remote control registers. But you're saying that remote control used to work fine for you?

**Reply by Vato9394** [reply]
no no, force charge and discharge never worked. I want to troubleshoot together with you to maybe enable this another way, maybe its another mosbus registery, I don't know.

**Reply by Vato9394** [reply]
remote control is disabled by default on the unit, I enabled remote control but it still wasn't working.
Maybe @PaRez280 did not enable this before trying to read the registers?

**Reply by canton7** [reply]
>  but it still wasn't working

What did you try, and what makes you think that it didn't work?

**Reply by Vato9394** [reply]
Failed to write registers is the error we get when selecting force charge/discharge

**Reply by canton7** [reply]
It looks like the inverter really doesn't support remote control, then?

**Reply by Vato9394** [reply]
It has an option on the menu on the inverter itself, and you can enable it...

**Reply by canton7** [reply]
Sure, but if we can't control it over modbus, then there's not a lot I can do.

**Reply by Vato9394** [reply]
But how are we sure that we cannot do it?

**Reply by canton7** [reply]
Because when I try, your inverter returns an error?

**Reply by Vato9394** [reply]
But maybe it is just a different modbus register?
How Can I find this? or is it just not possible?
I find it hard to understand that foxess is allowing it on H1 but not on H3

**Reply by canton7** [reply]
Well, they do support it on the H3. Just not the AIO-H3.

The remote control registers are normally exposed in the same block as things like Work Mode, which the AIO-H3 exposes in the normal place. It just doesn't seem to support specifically the remote control registers in that region.

You're welcome to try and find them elsewhere, but I'm not sure how you'd know you'd found them, in all honesty.

**Reply by PaRez280** [reply]
For me, version v1.10.2b3 did not display correctly only the entities battery Charge and battery discharge.  In version 1.11.1, everything is already OK.
